app/controllers/invoices_controller.rb

```diff
@@ -1,5 +1,11 @@
 class InvoicesController < ApplicationController
+  def index
+    @invoices = Invoice.all
+    @invoice_items = InvoiceItem.all
+    @merchant = Merchant.find(params[:merchant_id])
```

Comment:
is there a reason you need both invoice and invoice item here? i could be wrong. for revenue you can have a method in invoices that calls 

def total_revenue
    invoice_items.sum("unit_price * quantity")
  end